Sources/PointFree/SiteMiddleware.swift

```diff
@@ -11,103 +12,142 @@ import Styleguide
 import Tuple
 import Views
 
-public let siteMiddleware: Middleware<StatusLineOpen, ResponseEnded, Prelude.Unit, Data> =
-  requestLogger(logger: { Current.logger.log(.info, "\($0)") }, uuid: UUID.init)
-  <<< requireHerokuHttps(allowedInsecureHosts: allowedInsecureHosts)
-  <<< redirectUnrelatedHosts(isAllowedHost: { isAllowed(host: $0) }, canonicalHost: canonicalHost)
-  <<< router(notFound: routeNotFoundMiddleware)
-  <| currentUserMiddleware
-  >=> currentSubscriptionMiddleware
-  >=> render(conn:)
```

Comment:
I had a hard time wrapping the entire app stack in a `withDependency` for setting up the current user, route, etc. because it needed to be around both `render(conn:)` _and_ `route(notFound:)` (in order for 404's to show correct logged in state).

So, I unrolled all of this into a single async function. In some ways it's pretty nice, but once we get back to our server work I do think having a stack of middlewares to compose is nice.